hdr.go

```diff
@@ -69,23 +69,32 @@ func (h *Histogram) SetStartTimeMs(startTimeMs int64) {
 	h.startTimeMs = startTimeMs
 }
 
-// New returns a new Histogram instance capable of tracking values in the given
-// range and with the given amount of precision.
-func New(minValue, maxValue int64, sigfigs int) *Histogram {
-	if sigfigs < 1 || 5 < sigfigs {
-		panic(fmt.Errorf("sigfigs must be [1,5] (was %d)", sigfigs))
-	}
-
-	largestValueWithSingleUnitResolution := 2 * math.Pow10(sigfigs)
+// Construct a Histogram given the Lowest and Highest values to be tracked and a number of significant decimal digits.
+//
+// Providing a lowestDiscernibleValue is useful is situations where the units used for the histogram's values are
+// much smaller that the minimal accuracy required.
+// E.g. when tracking time values stated in nanosecond units, where the minimal accuracy required is a microsecond, the proper value for lowestDiscernibleValue would be 1000.
+func New(lowestDiscernibleValue, highestTrackableValue int64, numberOfSignificantValueDigits int) *Histogram {
+	if numberOfSignificantValueDigits < 1 || 5 < numberOfSignificantValueDigits {
+		panic(fmt.Errorf("numberOfSignificantValueDigits must be [1,5] (was %d)", numberOfSignificantValueDigits))
+	}
+
```

Comment:
add check on lowestDiscernibleValue must be >0
suggest forcing to 1 as min value

Reply:
@ahothan added the force to 1 as minimum discernible value. Regarding panic imho we should avoid extra panic calls within the code. Ideally we should deprecate `New()` in the future and enable a constructor that returns both the histogram and error. But for now I would stick to this constructor and only forcing 1 as a min value. wdyt?

Reply:
I agree we should never use panic inside a library.
I would suggest to also force the value within range and document accordingly - much better than panic.
Force to nearest boundary (1 if <1 or 5 if >5 would be appropriate).

Reply:
Agree. doing the change :+1: